lightning/src/ln/data_persister.rs

```diff
@@ -0,0 +1,27 @@
+//! Logic for persisting data from ChannelMonitors on-disk. Per-platform data
+//! persisters are separated into the lightning-persist-data crate.
+//! These objects mainly interface with the SimpleManyChannelMonitor when a
+//! channel monitor is added or updated, and when they are all synced on startup.
+use chain::channelmonitor::{ChannelMonitor, ChannelMonitorUpdate, ChannelMonitorUpdateErr};
+use chain::keysinterface::ChannelKeys;
+use chain::transaction::OutPoint;
+use std::collections::HashMap;
+
+/// ChannelDataPersister is responsible for persisting channel data: this could
+/// mean writing once to disk, and/or uploading to several backup services.
+pub trait ChannelDataPersister: Send + Sync {
+	/// The concrete type which signs for transactions and provides access to our channel public
+	/// keys.
+	type Keys: ChannelKeys;
+
+	/// Persist one channel's data. All backups should agree on a channel's state.
```

Comment:
I think that's the same requirement that the higher layer of channel monitor, previous channel's states are toxic. We should document that if the local data of this channel monitor you must be sure that downloading state from a backup is done at same commitment number.

Reply:
For availability reasons, you might only require a majority consensus among the backups, rather than all backups.

Reply:
Agree a majority is safe for backups.

Reply:
Gonna mark these comments as resolved but let me know if the docs could still use more information.